File: docs/snippets/python/xml-capture-for-replay_main.py
import asyncio
from pathlib import Path
from microsoft.agents.protocol import AgentProtocolClient


async def main() -> None:
    # Have a conversation via Client SDK
    client = AgentProtocolClient("http://localhost:5000")
    conversation = client.create_conversation()
    await conversation.send("Book a flight to Seattle")
    await conversation.send("The first one")

    # Export conversation instantly with str()
    xml = str(conversation)

    # Save or send to developer
    Path("customer-issue-456.xml").write_text(xml, encoding="utf-8")
    print("Exported conversation to customer-issue-456.xml")


if __name__ == "__main__":
    asyncio.run(main())

File: docs/snippets/csharp/xml-capture-for-replay_main.cs
using Microsoft.Agents.Protocol.Client;

// Have a conversation via Client SDK
var client = new AgentProtocolClient("http://localhost:5000");
var conversation = client.CreateConversation();
await conversation.SendAsync("Book a flight to Seattle").ConfigureAwait(false);
await conversation.SendAsync("The first one").ConfigureAwait(false);

// Export conversation instantly with ToString()
var xml = conversation.ToString();

// Save or send to developer
await File.WriteAllTextAsync("customer-issue-456.xml", xml).ConfigureAwait(false);
Console.WriteLine("Exported conversation to customer-issue-456.xml");
